src/test/java/com/kodilla/ecommercee/domain/CartTestSuite.java

```diff
@@ -0,0 +1,190 @@
+        Product product = new Product("blocks", "wooden blocks", new BigDecimal("4.50"));
+        group.getProducts().add(product);
+        product.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart();
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.setValue(new BigDecimal("22.50"));
+        List<Product> products = cart.getProducts();
+
+        //When
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //Then
+        assertNotEquals(0, cartId);
+        assertEquals(5, products.size());
+        assertEquals(cart.getProducts().get(0), cart.getProducts().get(3));
+        assertEquals("blocks", cart.getProducts().get(0).getProductName());
+    }
+
+    @Test
+    public void  shouldDifferentiateBetweenProductsWithSameNameDifferentPrice() {
+        //Given
+        Group group = new Group("Toys");
+        Product product1 = new Product("blocks", "wooden blocks", new BigDecimal("4.50"));
+        Product product2 = new Product("blocks", "wooden blocks", new BigDecimal("20"));
+        group.getProducts().add(product1);
+        group.getProducts().add(product2);
+        product1.getGroupsOfProduct().add(group);
+        product2.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart();
+        cart.getProducts().add(product1);
+        cart.getProducts().add(product2);
+        cart.setValue(new BigDecimal("24.50"));
+        List<Product> products = cart.getProducts();
+
+        //When
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //Then
+        assertNotEquals(0, cartId);
+        assertNotEquals(product1, product2);
+        assertEquals(2, products.size());
+    }
+
+    @Test
+    public void shouldRemoveAllProductsFromCart() {
+        //Given
+        Group group = new Group("Toys");
+        Product product1 = new Product("blocks", "wooden blocks", new BigDecimal("4.50"));
+        Product product2 = new Product("blocks", "wooden blocks", new BigDecimal("20"));
+        Product product3 = new Product("Teddy Bear", "little plush toy", new BigDecimal("10"));
+
+        group.getProducts().add(product1);
+        group.getProducts().add(product2);
+        group.getProducts().add(product3);
+
+        product1.getGroupsOfProduct().add(group);
+        product2.getGroupsOfProduct().add(group);
+        product3.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart();
+        cart.getProducts().add(product1);
+        cart.getProducts().add(product2);
+        cart.getProducts().add(product3);
+
+        cart.setValue(new BigDecimal("34.50"));
+        List<Product> products = cart.getProducts();
+
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //When
+        cart.getProducts().removeAll(products);
```

Comment:
Tutaj nie sprawdzasz usuwania produktów z koszyka w bazie, tylko usuwanie produktów z obiektu. Musisz potem taki obiekt zapisać, potem wczytać z bazy i dopiero sprawdzić czy są tam produkty

Reply:
ale skoro Hibernate monitoruje zmiany w obiektach i jeśli jakieś były sam je zapisuje do bazy i nie trzeba robić save(), to chyba nie ma różnicy którą listę się sprawdza?